include/circt/Dialect/FIRRTL/FIRParser.h

```diff
@@ -107,10 +107,10 @@ struct FIRVersion {
   uint16_t patch;
 };
 
-constexpr FIRVersion minimumFIRVersion(0, 2, 0);
+constexpr FIRVersion minimumFIRVersion(2, 0, 0);
 constexpr FIRVersion nextFIRVersion(3, 3, 0);
 constexpr FIRVersion exportFIRVersion(4, 0, 0);
-constexpr FIRVersion defaultFIRVersion(1, 0, 0);
+constexpr FIRVersion defaultFIRVersion(2, 0, 0);
```

Comment:
This is a reasonable change for this PR.

Given the fact that removing partial connect support means that we can drop the notion of a default version because:

1. `firtool` can now only parse FIRRTL versions 2.0.0 or later.
2. The spec says that FIRRTL versions before 1.1.0 (the first version which added versions can leave off the version string). All others have to include it. (The language here could be more direct, i.e., _should_, etc.)

Thereby, given (1), we can mandate checking for the version string and drop a default version.

WDYT?

Reply:
I think that's the way to go.

Part of me wants to keep the default (and bump it) purely out of convenience for writing examples 🙃 , but that's minor and not what the spec says 👍 .